torch/numa/binding.py

```diff
@@ -40,146 +42,184 @@ class AffinityMode(str, Enum):
-                f"Affinity mode {numa_options.affinity_mode} not supported."
-            )
 
-        if numa_options.should_fall_back_if_binding_fails:
-            _raise_if_numactl_fails_dry_run(numactl_options=numactl_command_options)
+        numactl_options = _get_numactl_cli_options(
+            command_args=command_args, gpu_index=gpu_index, numa_options=numa_options
+        )
+        logger.info("Computed numactl_options=%r", numactl_options)
+
+        _raise_if_numactl_fails_dry_run(numactl_options=numactl_options)
+        logger.info("Validated numactl_options=%r", numactl_options)
+
+        full_numactl_command = _get_assembled_command_from_pieces(
+            command_args=command_args, numactl_options=numactl_options
+        )
+        logger.info(
+            "Successfully wrapped command with numa_bindings. Returning %r",
+            full_numactl_command,
+        )
         signpost_event(
             category="numa_binding",
             name="wrap_command_success",
-            parameters={
-                "original_command_args": command_args,
-                "gpu_index": gpu_index,
-                "numa_options": numa_options,
-                "numactl_command_options": numactl_command_options,
-            },
+            parameters={**kwargs, "result": full_numactl_command},
         )
-        return numactl_command_options
+        return full_numactl_command
     except Exception:
         signpost_event(
             category="numa_binding",
             name="wrap_command_exception",
             parameters={
+                **kwargs,
                 "traceback": traceback.format_exc(),
-                "original_command_args": command_args,
-                "gpu_index": gpu_index,
-                "numa_options": numa_options,
             },
         )
         logger.exception(
-            """Failed to wrap command with NUMA bindings.
-            Input:
-                command_args=%r,
-                gpu_index=%d,
-                numa_options=%r,
-        """,
-            command_args,
-            gpu_index,
-            numa_options,
+            "Failed to wrap command with NUMA bindings for input = %r", kwargs
         )
+        if numa_options.should_fall_back_if_binding_fails:
+            logger.warning("Falling back to original command without NUMA bindings.")
+            return None
         raise
 
 
+def _get_temporary_executable_for_command(
+    *,
+    command_args: tuple[str, ...],
+) -> str:
+    """
+    Returns:
+        Path to a temporary file which executes the specified command. The executable
+        deletes itself the first time it runs, so do not try to run it multiple times.
+    """
+    fd, path = tempfile.mkstemp(
+        prefix="pytorch-numa-bind",
+        suffix=".sh",
+    )
+
+    # We do rm first to guarantee the file deletes itself. The rest of the file
+    # will still run as intended.
+    contents = f"""#!/bin/bash
+rm -- "$0"
```

Comment:
maybe add a comment if people see these files laying around what this is for/what created it?